Fabric.Identity.API/Services/IdPSearchServiceProvider.cs

```diff
@@ -1,161 +1,161 @@
-                    return null;
-                }
-
-                var result = JsonConvert.DeserializeObject<UserSearchResponse>(responseContent);
-
-                return new ExternalUser
-                {
-                    FirstName = result.FirstName,
-                    LastName = result.LastName,
-                    MiddleName = result.MiddleName,
-                    SubjectId = result.SubjectId,
-                    Email = result.Email
-                };
-            }
-            catch (TaskCanceledException e)
-            {
-                _logger.Error($"The request to the search service was canceled: {e.Message}");
-                throw;
-            }
-            catch (HttpRequestException e)
-            {
-                var baseException = e.GetBaseException();
-
-                _logger.Error($"There was an error connecting to the search service: {baseException.Message}");
-                throw;
-            }
-        }
-
-        public ICollection<ExternalUser> SearchUsers(string searchText)
-        {
-            throw new NotImplementedException();
-        }
-    }
-
-    public class UserSearchResponse
-    {
-        public string SubjectId { get; set; }
-        public string FirstName { get; set; }
-        public string MiddleName { get; set; }
-        public string LastName { get; set; }
-        public string Email { get; set; }
-        public string PrincipalType { get; set; }
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Fabric.Identity.API.Configuration;
+using Fabric.Identity.API.Infrastructure;
+using Fabric.Identity.API.Models;
+using Fabric.Identity.API.Extensions;
+using Fabric.Platform.Shared.Exceptions;
+using IdentityModel.Client;
+using Newtonsoft.Json;
+using Polly.CircuitBreaker;
+using Serilog;
+
+namespace Fabric.Identity.API.Services
+{
+    public class IdPSearchServiceProvider : IExternalIdentityProviderService
+    {
+        private readonly IAppConfiguration _appConfig;
+        private readonly ILogger _logger;
+        private readonly PolicyProvider _policyProvider;
+        private readonly HttpClient _httpClient;
+        private readonly IHttpRequestMessageFactory _httpRequestMessageFactory;
+
+        public IdPSearchServiceProvider(IAppConfiguration appConfig, ILogger logger, PolicyProvider policyProvider, HttpClient httpClient, IHttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _appConfig = appConfig;
+            _logger = logger;
+            _policyProvider = policyProvider;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        public async Task<ExternalUser> FindUserBySubjectId(string subjectId)
+        {
+            if (!_appConfig.IdentityProviderSearchSettings.IsEnabled)

```

Comment:
Is this part of the port over?

Seems strange that only this search has a check to see if it's disabled, it's probably better to inject an empty IExternalIdentityProviderService during startup

Reply:
This class will probably be removed in the next couple of versions. The current search endpoint does reference this, although it might be broken (which is one of the reasons why we wanted to migrate IdPSS over in the first place).